src/crimson/common/fatal_signal.cc

```diff
@@ -62,13 +65,59 @@ static void print_backtrace(std::string_view cause) {
   //       see handle_fatal_signal()
 }
 
-static void print_segv_info(const siginfo_t* siginfo)
+static void print_segv_info(const siginfo_t& siginfo)
 {
-  std::cerr << "Fault at location: " << siginfo->si_addr << std::endl;
+  std::cerr \
+     << "Dump of siginfo:" << std::endl
+     << "  si_signo: " << siginfo.si_signo << std::endl
+     << "  si_errno: " << siginfo.si_errno << std::endl
+     << "  si_code: " << siginfo.si_code << std::endl
+     << "  si_pid: " << siginfo.si_pid << std::endl
+     << "  si_uid: " << siginfo.si_uid << std::endl
+     << "  si_status: " << siginfo.si_status << std::endl
+     << "  si_utime: " << siginfo.si_utime << std::endl
+     << "  si_stime: " << siginfo.si_stime << std::endl
+     << "  si_int: " << siginfo.si_int << std::endl
+     << "  si_ptr: " << siginfo.si_ptr << std::endl
+     << "  si_overrun: " << siginfo.si_overrun << std::endl
+     << "  si_timerid: " << siginfo.si_timerid << std::endl
+     << "  si_addr: " << siginfo.si_addr << std::endl
+     << "  si_band: " << siginfo.si_band << std::endl
+     << "  si_fd: " << siginfo.si_fd << std::endl
+     << "  si_addr_lsb: " << siginfo.si_addr_lsb << std::endl
+     << "  si_lower: " << siginfo.si_lower << std::endl
+     << "  si_upper: " << siginfo.si_upper << std::endl
+     << "  si_pkey: " << siginfo.si_pkey << std::endl
+     << "  si_call_addr: " << siginfo.si_call_addr << std::endl
+     << "  si_syscall: " << siginfo.si_syscall << std::endl
+     << "  si_arch: " << siginfo.si_arch << std::endl;
   std::cerr << std::flush;
 }
 
-void FatalSignal::signaled(const int signum, const siginfo_t* siginfo)
+static void print_proc_maps()
+{
+  const int fd = open("/proc/self/maps", O_RDONLY|O_BINARY);
+  if (fd < 0) {
+    std::cerr << "can't open /proc/self/maps. procfs not mounted?" << std::endl;
+    return;
+  }
+  std::cerr << "Content of /proc/self/maps:" << std::endl;
+  while (true) {
+    char chunk[4096] = {0, };
+    const ssize_t r = safe_read(fd, chunk, sizeof(chunk));
+    if (r < 0) {
+      std::cerr << "error while reading /proc/self/maps: " << r << std::endl;
+      return;
+    } else {
+      std::cerr << chunk << std::flush;
```

Comment:
do we have access to `stderr` in a daemon?

Reply:
Looks like so as the previously printed `Fault at location:` was present in the logs:

```
rzarzynski@teuthology:/home/teuthworker/archive/rzarzynski-2021-06-08_09:11:05-rados-master-distro-basic-smithi/6159602$ less ./remote/smithi057/log/ceph-osd.7.log.gz
...
Fault at location: 0x98
```